### PR TITLE
Cmssw 7 6 x pixel lumi dqm

### PR DESCRIPTION
**Pull request by rpatkennyiii**
This an attempt to resurrect the PixelLumi DQM code that was lost in the CVS to git migration.

### CODE REVIEW

File: DQM/PixelLumi/plugins/BuildFile.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+<use name="Geometry/TrackerGeometryBuilder"/>
+<use name="Geometry/Records"/>
+<use name="CommonTools/UtilAlgos"/>
+<use name="CondFormats/SiPixelObjects"/>
+<use name="HLTrigger/HLTcore"/>
+<use name="CondFormats/RunInfo"/>
+<use name="CondTools/L1Trigger"/>
+<use name="DQMServices/Core"/>
+<use name="RecoLuminosity/LumiProducer"/>
+<use name="clhep"/>
```
Comment (deguio) [inline DQM/PixelLumi/plugins/BuildFile.xml]: please make some cleanup here.
most of the packages are not needed

File: DQM/PixelLumi/plugins/PixelLumiDQM.h [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Provenance/interface/RunID.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+
+#include "TObject.h"
+
+class TTree;
```
Comment (deguio) [inline DQM/PixelLumi/plugins/PixelLumiDQM.h]: please remove

File: DQM/PixelLumi/plugins/PixelLumiDQM.h [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+class TH2D;
```
Comment (deguio) [inline DQM/PixelLumi/plugins/PixelLumiDQM.h]: please remove

File: DQM/PixelLumi/plugins/PixelLumiDQM.h [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    std::vector<double> dnumFM;
+    std::vector<double> dnumFP;
+  private:
+  };
+
+  edm::InputTag fVtxLabel;
+  edm::InputTag fPixelClusterLabel;
+  edm::InputTag fGtEvmLabel;
+
+  TTree* fTree;
```
Comment (deguio) [inline DQM/PixelLumi/plugins/PixelLumiDQM.h]: not used

File: DQM/PixelLumi/python/AlCaLumiPixelsZeroBiasSelector_cfi.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import FWCore.ParameterSet.Config as cms
```
Comment (deguio) [inline DQM/PixelLumi/python/AlCaLumiPixelsZeroBiasSelector_cfi.py]: the whole file can be removed. the defined filter is never used.

File: DQM/PixelLumi/python/PixelLumiDQM_cfi.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                                 includeTrackInfo = cms.untracked.bool(False),
+                                 includeStripClusterInfo = cms.untracked.bool(False),
+                                 includePixelClusterInfo = cms.untracked.bool(True),
+                                 includePixelQualCheckHistos = cms.untracked.bool(True),
+                                 #parameters to connect to db (from L1TMonitor)
+                                 oracleDB   = cms.string("oracle://cms_orcon_prod/cms_runtime_logger"),
+                                 pathCondDB = cms.string(""),
+#                                 pathCondDB = cms.string("/nfshome0/popcondev/conddb"),        
+                                 # This is the correct list of modules to be ignored for 2012.
+                                 deadModules = cms.untracked.vuint32(302059800,
```
Comment (deguio) [inline DQM/PixelLumi/python/PixelLumiDQM_cfi.py]: isn't this information available from DB?

File: DQM/PixelLumi/python/RandomTriggerSelector_cfi.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import FWCore.ParameterSet.Config as cms
+
+# Selects only random-trigger events.
+
+random_trigger_selector = cms.EDFilter("HLTHighLevel",
```
Comment (deguio) [inline DQM/PixelLumi/python/RandomTriggerSelector_cfi.py]: the whole file can be removed. filter not used anywhere.

**Comment by mojoe137** [comment]
I think I've dealt with all the issues. As for DB access to the dead channels, the code I inherited used this method, putting in the dead modules in a list in the cfi. I'll check with the others in the LUMI POG to see if there is reason for this.

**Comment by deguio** [comment]
thanks @mojoe137 
one last thing is to move the module to use the consumes. the getBylLabel should be changed to getByToken and the consumes should be declared.
here an example on how to migrate the code:
https://github.com/cms-sw/cmssw/blob/CMSSW_7_6_X/DQMServices/Examples/interface/DQMExample_Step1.h
https://github.com/cms-sw/cmssw/blob/CMSSW_7_6_X/DQMServices/Examples/src/DQMExample_Step1.cc

once this is done I believe we are ready to go. you should submit the final version of this PR to 74x and 75x and update the twiki at:
https://twiki.cern.ch/twiki/bin/viewauth/CMS/DQMP5TagCollector

so that we can test and deploy online.
thanks,
F.

@danduggan

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
status:
this PR is being tested on the playback and will be deployed in production soon (later today/tomorrow) given the fact that it is urgent.
however there are a couple of things to be fixed before we go with a merge:
- please update the PixelLumiDQM to take the list of deadModules from DB (the current hardcoded list is out of date)
- please replace the filter ZeroBiasSelector_cfi.py with an option at the source directly in the pixellumi_dqm_sourceclient-live_cfg.py config [1]



[1]
add one line like this:
process.source.SelectEvents = cms.untracked.vstring('HLT_*Mu*')

if you want to be more specific, you can OR things:
process.source.SelectEvents = cms.untracked.vstring('HLT_*SingleMu*', 'HLT_*DoubleMu*')

**Comment by deguio** [comment]
let's converge on this one first then the last needed changes can be backported

**Comment by mojoe137** [comment]
There is come code here: https://github.com/cms-sw/cmssw/blob/68b4ccfc9d6f7b0ae0980969622686957e61fc9a/SimGeneral/TrackingAnalysis/src/PixelPSimHitSelector.cc#L35, which seems to access the dead modules from the db.

**Comment by mojoe137** [comment]
I poked around to try and find the most recent list of bad pixel modules. What I found on  cms_orcon_adg database was SiPixelQuality_v24 tag. If I use what I find there then I would change the code starting here https://github.com/mojoe137/cmssw/blob/CMSSW_7_4_X-PixelLumiDQM/DQM/PixelLumi/python/PixelLumiDQM_cfi.py#L8 with

deadModules = cms.untracked.vuint32(302057496, 302059292, 302188820, 302123296, 302127136, 302126364, 302126624, 302194200, 302057732, 302124292, 302124296, 302124300, 302124304, 302124548, 302124552, 302124556, 302124560),

This is probably closer to what the final list will be, but I this the eventual goal would be to have a list that was consistent over a long run period.

**Comment by danduggan** [comment]
Hey @mojoe137, please cross check this list with the pixel community (not in their code but contact them for confirmation), then update the 74X PR accordingly. The 76X configuration should instead be reading this from the DB, so only update this one when it's in that configuration. Again, the pixel community should be able to help.

**Comment by danduggan** [comment]
Hey @mojoe137, thanks for following up and confirming that you can run without the mask. I think all points have been addressed. The only final thing is the actual trigger you would need to select - as it is now in the code, "AlCa_LumiPixels_ZeroBias*" will select identically zero events in online. My suggestion would be to make this "HLT_ZeroBias*" for the moment and iterate with the lumi pog in case other changes are needed. Please also propagate all updates to your 74x and 75x PRs too.

@deguio, I'm happy once this is done - I'll let you sign-off.

**Comment by deguio** [comment]
+1

**Comment by deguio** [comment]
thanks @mojoe137 
please don't forget to propagate to 75 and 74 all the changes otherwise at the next release update they will be lost.
ciao,
F.

**Comment by mojoe137** [comment]
@danduggan, @deguio, is there something I need to do to get this wrapped up? Last I talked to Dan he said he didn't think it was an issue with my code, but some other part of the dqm package.

**Comment by deguio** [comment]
ciao @mojoe137 
please have a look at  #10074 
there is something to fix.
thanks,
F.

**Comment by mojoe137** [comment]
it looked like it didn't like the d suffix for the double constants, so I took them out. When I compiled it myself it didn't complain about this, but I suppose it doesn't hurt to remove them.

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
+1
online only

**Pull request merged**